src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -205,48 +444,82 @@ static uct_iface_ops_t uct_gga_mlx5_iface_tl_ops = {
-    .ep_get_address           = ucs_empty_function_return_unsupported,
+    .ep_create                = UCS_CLASS_NEW_FUNC_NAME(uct_gga_mlx5_ep_t),
+    .ep_destroy               = UCS_CLASS_DELETE_FUNC_NAME(uct_gga_mlx5_ep_t),
+    .ep_get_address           = uct_gga_mlx5_ep_get_address,
     .ep_connect_to_ep         = ucs_empty_function_return_unsupported,
-    .iface_flush              = ucs_empty_function_return_unsupported,
-    .iface_fence              = (uct_iface_fence_func_t)ucs_empty_function_do_assert,
-    .iface_progress_enable    = ucs_empty_function_do_assert_void,
-    .iface_progress_disable   = ucs_empty_function_do_assert_void,
+    .iface_flush              = uct_rc_iface_flush,
+    .iface_fence              = uct_rc_iface_fence,
+    .iface_progress_enable    = uct_rc_mlx5_iface_progress_enable,
+    .iface_progress_disable   = uct_base_iface_progress_disable,
     .iface_progress           = (uct_iface_progress_func_t)ucs_empty_function_do_assert,
-    .iface_event_fd_get       = ucs_empty_function_return_unsupported,
-    .iface_event_arm          = ucs_empty_function_return_unsupported,
+    .iface_event_fd_get       = uct_gga_mlx5_iface_event_fd_get,
+    .iface_event_arm          = uct_gga_mlx5_iface_arm,
     .iface_close              = uct_gga_mlx5_iface_t_delete,
-    .iface_query              = (uct_iface_query_func_t)ucs_empty_function_do_assert,
+    .iface_query              = uct_gga_mlx5_iface_query,
     .iface_get_address        = ucs_empty_function_return_success,
-    .iface_get_device_address = ucs_empty_function_return_unsupported,
+    .iface_get_device_address = uct_gga_mlx5_iface_get_device_address,
     .iface_is_reachable       = uct_base_iface_is_reachable
 };
 
+static int
+uct_gga_mlx5_iface_is_reqchable_v2(
+        const uct_iface_h tl_iface, const uct_iface_is_reachable_params_t *params)
+{
+    uct_ib_iface_t *iface                   = ucs_derived_of(tl_iface,
+                                                             uct_ib_iface_t);
+    uct_ib_device_t *device                 = uct_ib_iface_device(iface);
+    const uct_gga_mlx5_dev_addr_t *dev_addr = (const uct_gga_mlx5_dev_addr_t *)
+            UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD, params, device_addr,
+                            DEVICE_ADDR, NULL);
+    uct_iface_is_reachable_params_t ib_params;
+
+    if ((dev_addr == NULL) ||
+        (be64toh(dev_addr->be_sys_image_guid) !=
+         be64toh(device->dev_attr.orig_attr.sys_image_guid))) {
+        return 0;
+    }
+
+    ib_params             = *params;
+    ib_params.device_addr = (const uct_device_addr_t*)&dev_addr->ib_addr;
+    return uct_ib_iface_is_reachable_v2(tl_iface, &ib_params);
+}
+
+static void
+uct_gga_mlx5_iface_qp_cleanup(uct_rc_iface_qp_cleanup_ctx_t *ctx)
+{
+    uct_gga_mlx5_iface_qp_cleanup_ctx_t *gga_ctx =
+            ucs_derived_of(ctx, uct_gga_mlx5_iface_qp_cleanup_ctx_t);
+
+    ibv_dereg_mr(gga_ctx->dma_opaque.mr);
```

Comment:
maybe uct_ib_dereg_mr?

Reply:
should be aligned with https://github.com/openucx/ucx/pull/9845/files#r1587560718